packages/ui/src/2_molecules/WalletIdentity/components/AddressBadge/AddressBadge.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { FC } from 'react';
+
+import blockies from 'ethereum-blockies';
+
+import { prettyTx } from '../../../../utils';
+
+export type AddressBadgeProps = {
+  address: string;
+  startLength?: number;
+  endLength?: number;
+};
+
+export const AddressBadge: FC<AddressBadgeProps> = ({
+  address,
+  startLength = 6,
+  endLength = 4,
+}) => {
+  const getWalletAddrBlockieImg = (): string => {
+    return blockies
+      .create({
+        // All options are optional
+        seed: address, // seed used to generate icon data, default: random
+        color: '#dfe', // to manually specify the icon color, default: random
+        bgcolor: '#aaa', // choose a different background color, default: random
+        size: 8, // width/height of the icon in blocks, default: 8
+        scale: 3, // width/height of each block in pixels, default: 4
+        spotcolor: '#000', // each pixel has a 13% chance of being of a third color,
+      })
+      .toDataURL();
+  };
+
+  return (
+    <span className="flex flex-nowrap flex-row items-center w-full truncate font-semibold">
+      <span>
+        <img
+          className="rounded-full mr-2"
+          src={getWalletAddrBlockieImg()}
+          alt="wallet address"
+        />
+      </span>
+      <span>{prettyTx(address || '', startLength, endLength)}</span>
+    </span>
```

Comment:
add custom classes to css modules instead of using tailwindcss directly in the component